db/migrate/20230730013746_downcase_user_emails.rb

```diff
@@ -0,0 +1,11 @@
+class DowncaseUserEmails < ActiveRecord::Migration[6.1]
+  # normalize all emails to lowercase. This may fail if there are duplicates
+  # which need to be handled manually
+  def up
+    safety_assured do
+      execute <<~SQL
+        UPDATE users SET email = LOWER(email) WHERE deleted_at IS NULL
```

Comment:
Might be worth updating deleted users too, except if you think that'll create an issue because of duplicate emails, which is probably why this is here....